src/lib.rs

```diff
@@ -67,3 +67,33 @@ unsafe extern "C" fn clock_gettime(
 
     retval
 }
+
+#[no_mangle]
+unsafe extern "C" fn getrandom(
+    buf: *mut libc::c_void,
+    mut buflen: libc::size_t,
+    flags: libc::c_uint,
+) -> libc::ssize_t {
+    // TODO: is this needed? Maybe just `buflen = buflen.min(libc::ssize_t::MAX)` ?
```

Comment:
This number chosen based on https://man7.org/linux/man-pages/man2/getrandom.2.html#NOTES

Reply:
I there is the GRND_RANDOM flag for using `random` instead of `urandom`(under linux). Maybe we could check those to change the maximum buffer length(for `urandom` it's 33554431, while for `random` it's 512), so it would more closely resemble the original.

Reply:
Interesting... as it turns out, `getrandom`, `getentropy`, `GRND_NONBLOCK`, and `GRND_RANDOM` seem to only be defined in newlib for Cygwin, and not on 3DS (or any other platform).

So, I guess I will just add them to `horizon/mod.rs` in `libc`, and anyone using it will just have to live with the fact that `GRND_NONBLOCK` doesn't work at all.